docs/site/Req-res-cycle.md

```diff
@@ -0,0 +1,232 @@
+the APIs that create the endpoints on the server. We will then follow the path
+taken by a request, to see how it makes its way through the various parts of the
+framework to return a result.
+
+### Setting up the request-responses infrastruture
+
+The endpoints on a LoopBack app can be categorized into controller endpoints and
+non-controller endpoints. Controller endpoints are those that are created by
+LoopBack controller methods; non-controller endpoints are those that are created
+by other APIs.
+
+#### Controller endpoints
+
+Controller methods decorated with operation decorators like `@get()`, `@post()`,
+`@put` etc., create endpoints on the app for the corresponding HTTP verbs. The
+behavior of these endpoints are entirely dependent on the implementation of the
+controller method.
+
+{% include tip.html content="Apart from controller files in the `controllers` directory,
+controllers may be added to the app by [components](https://loopback.io/doc/en/lb4/Components.html)." %}
+
+In the request-response cycle section we will see how implemenation details
+determine the course of a request to these endpoints - they may or may not actually
+interact with a model.
+
+#### Non-controller endpoints
+
+The following APIs can create non-controller endpoints.
+
+##### 1. app.static(path, rootDir, options)
+
+This method maps to the underlying Express' [static](https://expressjs.com/en/4x/api.html#express.static)
+method. It creates dynamic endpoints for files in the `rootDir` to be served
+at `path`.
+
+More details at [RestApplication.static()](https://loopback.io/doc/en/lb4/apidocs.rest.restapplication.static.html)
+doc.
+
+##### 2. app.redirect(fromPath, toPathOrUrl, statusCode)
+
+This method redirects requests to `fromPath` to `toPathOrUrl` with the status
+code `statusCode`.
+
+More details at [RestApplication.redirect()](https://loopback.io/doc/en/lb4/apidocs.rest.restapplication.redirect.html)
+doc.
+
+##### 3. app.mountExpressRouter(basePath, router, spec)
+
+This method mounts an Express [router](https://expressjs.com/en/4x/api.html#router)
+in the LoopBack app. You can use it for adding custom endpoints in your app using
+the Express middleware format.
+
+More details at [RestApplication.mountExpressRouter()
+](https://loopback.io/doc/en/lb4/apidocs.rest.restapplication.mountexpressrouter.html) doc.
+
+##### 4. non-controller endpoints added by components
+
+[Components](https://loopback.io/doc/en/lb4/Components.html) are the extension
+units of LoopBack. Since they can access the LoopBack app instance, the request
+and response instances using [dependency injection](https://loopback.io/doc/en/lb4/Dependency-injection.html),
+they are capable of adding new endpoints and determining the result of a request.
+
+### The request-response cycle
+
+The request-response cycle involves many components, especially if the request
+is to a controller-endpoint which interacts with a model.
```

Comment:
`the request is to a controller-endpoint` -> `the request is coming from a controller-endpoint`?

Reply:
Made it "interacts with the database".